ui/app/src/Utils/FormValidation.elm

```diff
@@ -0,0 +1,62 @@
+module Utils.FormValidation
+    exposing
+        ( ValidationResult(Valid, NotValid)
+        , ValidatedString
+        , validateDate
+        , areValidResults
+            , validatedMatcherToMatcher
+        , ValidatedMatcher
+        )
+
+import Utils.Date exposing (timeFromString)
+import Silences.Types exposing (Silence)
+import Utils.Types exposing (Matcher)
+
+
+type ValidationResult
+    = Valid
+    | NotValid String
+
+
+type alias ValidatedString =
+    { value : String
+    , result : ValidationResult
+    }
```

Comment:
@mxinden you would still need to store the raw input value + a result of validated input separately. The idea is to keep the validation result, because it's type may be not the raw string type, e.g. for a time input. Then, because each field stores the validation result, we can get all values from there and create a silence out of them, maybe via mapping all results with a silence constructor using smth like http://package.elm-lang.org/packages/elm-lang/core/latest/Result#map5